internal/cli/universal_login_templates.go

```diff
@@ -0,0 +1,454 @@
+	"errors"
+	"fmt"
+	"net"
+	"net/http"
+	"net/url"
+	"path/filepath"
+	"text/template"
+	"time"
+
+	"github.com/auth0/go-auth0/management"
+	"github.com/fsnotify/fsnotify"
+	"github.com/guiguan/caster"
+	"github.com/pkg/browser"
+	"github.com/spf13/cobra"
+	"golang.org/x/sync/errgroup"
+
+	"github.com/auth0/auth0-cli/internal/ansi"
+	"github.com/auth0/auth0-cli/internal/auth0"
+	"github.com/auth0/auth0-cli/internal/prompt"
+)
+
+const (
+	defaultPrimaryColor    = "#0059d6"
+	defaultBackgroundColor = "#000000"
+	defaultLogoURL         = "https://cdn.auth0.com/manhattan/versions/1.3935.0/assets/badge.png"
+)
+
+var (
+	errNotAllowed = errors.New(
+		"this feature requires at least one custom domain to be set and verified " +
+			"for the tenant, use 'auth0 domains create' to create one and 'auth0 domains verify' to have it verified",
+	)
+
+	//go:embed data/branding/storybook/*
+	templatePreviewAssets embed.FS
+
+	//go:embed data/branding/tenant-data.js
+	tenantDataAsset string
+
+	//go:embed data/branding/default-template.liquid
+	templateBasic string
+
+	//go:embed data/branding/image-template.liquid
+	templateWithImage string
+
+	//go:embed data/branding/footer-template.liquid
+	templateWithFooter string
+
+	templateBody = Flag{
+		Name:       "Template",
+		LongForm:   "template",
+		ShortForm:  "t",
+		Help:       "Custom page template for Universal Login.",
+		IsRequired: true,
+	}
+
+	templateOptions = pickerOptions{
+		{"Basic template", templateBasic},
+		{"Template with login box and background image", templateWithImage},
+		{"Template with page footers", templateWithFooter},
+	}
+)
+
+// TemplateData contains all the variables we project onto our embedded go
+// template. These variables largely resemble the same ones in the auth0
+// branding template.
+type TemplateData struct {
+	Filename        string
+	Clients         []ClientData
+	PrimaryColor    string
+	BackgroundColor string
+	LogoURL         string
+	TenantName      string
+	Body            string
+}
+
+// ClientData is a minimal representation of an Auth0 Client as defined in the
+// management API. This is used within the branding machinery to populate the
+// tenant data.
+type ClientData struct {
+	ID      string `json:"id"`
+	Name    string `json:"name"`
+	LogoURL string `json:"logo_url,omitempty"`
+}
+
+func showBrandingTemplateCmd(cli *cli) *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "show",
+		Args:  cobra.NoArgs,
+		Short: "Display the custom template for Universal Login",
+		Long:  "Display the custom template for the Universal Login experience.",
+		Example: `  auth0 universal-login templates show
+  auth0 ul templates show`,
+		RunE: func(cmd *cobra.Command, args []string) error {
+			var currentTemplate *management.BrandingUniversalLogin
+			if err := ansi.Waiting(func() (err error) {
+				currentTemplate, err = cli.api.Branding.UniversalLogin()
+				if err != nil {
+					if mErr, ok := err.(management.Error); ok && mErr.Status() == http.StatusNotFound {
```

Comment:
Eventually we could extract this logic somewhere, as this pattern of checking for error 404 (Management API quirk) has popped up repeatedly.